test/regressions/index.js

```diff
@@ -1,5 +1,5 @@
 import * as React from 'react';
-import ReactDOM from 'react-dom';
+import * as ReactDOM from 'react-dom';
```

Comment:
@m4theushw When do we plan to use React 18? It would then be 

```suggestion
import * as ReactDOM from 'react-dom/client';
```

per https://reactjs.org/blog/2022/03/29/react-v18.html#new-client-and-server-rendering-apis

Reply:
I'm only waiting for the core repo to adopt React 18 in the infra. But nothing stops us from adopting earlier. https://github.com/mui/mui-x/pull/4155 is open.